### DISCUSSION TITLE
[4.0] Joomla 4 - Extremely CPU intensive (Server load)

### DISCUSSION THREAD

**Discussion by HRIT-Florian** [joomla/joomla-cms#33142] [General]
### Steps to reproduce the issue
Install blank Joomla4 Beta7 on a server with CPU monitoring and without any other installation.

### Expected result
These ist for example a productive Joomla3 CPU activity on a website load

![grafik](https://user-images.githubusercontent.com/12803406/114772195-d0805700-9d6d-11eb-9191-0a999d52afe6.png)

### Actual result
And Joomla4, when just opening backend login or any other backend list view or something else.
![grafik](https://user-images.githubusercontent.com/12803406/114772443-163d1f80-9d6e-11eb-9547-8bafb26a4d20.png)

It also effects any view, but high load was deteced on component list views, login view and edit views.

We have tested it on high end Servers with different PHP Versions.
Anytime same result.
Tested also by disabling any Joomla Core Plugin

Note: Problem is not noticeable in the browser and also not in the JDebug console!

But if more clients at same time open a stream, the Server goes down and than ist's really noticeable with a Service Unavailable Error! This must not happen, not with a bare Joomla installation on a high-end machine!

We have debugged with exit, etc. and could just reduce the server load minimal, but not to any acceptable value.
Maybe the problem depends on huge file dependency, not sure!

### System information (as much as possible)
**Tested on: Intel Xeon E3-1270v6 3,8 GHz 4 (8) Core (Dedicated Server)**
 Setting | Value
--|--
PHP Built On | Linux s3334 4.19.44-gentoo-mw1 1 SMP Tue May 21 14:18:35 CEST 2019 x86_64
Database Type | mysql
Database Version|5.7.25-28
PHP Version |7.3.25,  7.3.25 and 8.0 latest FPM
Web Server |Apache
WebServer to PHP Interface |fpm-fcgi
Joomla! Version |Joomla! 4.0.0-beta7 Beta [ Mañana ] 2-February-2021 19:23 GMT

**Comment by Quy** [comment]
Please test with the nightly build to see if it is still an issue. Thanks.

https://developer.joomla.org/nightly-builds.html

**Comment by HRIT-Florian** [comment]
@Quy same, tested again with current nightly build **Extremely CPU intensive !!**

![grafik](https://user-images.githubusercontent.com/12803406/114776657-0247ec80-9d73-11eb-8301-6becaff26ae8.png)

Joomla! Version | Joomla! 4.0.0-beta8-dev Development [ Mañana ] 2-February-2021 20:02 GMT

**Each test ist one client website click (single page load)!**

**Comment by dgrammatiko** [comment]
Share your exact PHP info. I'll bet you're missing something and a polyfill is used thus the bad performance

**Comment by HRIT-Florian** [comment]
@dgrammatiko attached export settings from metioned joomla instance (including php info)
[systeminfo-2021-04-14T23 48 39+02 00.zip](https://github.com/joomla/joomla-cms/files/6313895/systeminfo-2021-04-14T23.48.39%2B02.00.zip)

**Comment by dgrammatiko** [comment]
maybe it's easier to debug this on your side. My local setup has these extensions enabled:

```bash
Dimitriss-MBP:joomla-cms dgrammatiko$ php -m
[PHP Modules]
bcmath
bz2
calendar
Core
ctype
curl
date
dom
exif
fileinfo
filter
gd
gettext
hash
iconv
imap
intl
json
ldap
libxml
mbstring
mysqli
mysqlnd
openssl
pcntl
pcre
PDO
pdo_mysql
pdo_pgsql
pdo_sqlite
pgsql
Phar
posix
readline
Reflection
session
SimpleXML
soap
sockets
sodium
SPL
sqlite3
standard
tokenizer
xdebug
xml
xmlreader
xmlwriter
xsl
zip
zlib

[Zend Modules]
Xdebug

```

**Comment by Fedik** [comment]
Please try apply the changes from that PR #31143 then disable "Log Deprecated" in global config.
And tell about results.

The graph itself does not tell anything usfull that we can use for track the issue.
~But would be interesting see graph compassion J3 vs J4 on same server.~

**Comment by brianteeman** [comment]
> But would be interesting see graph compassion J3 vs J4 on same server.

First post

**Comment by Fedik** [comment]
but I read it :smile:

well then, ignore my last sentence :smile:

**Comment by HRIT-Florian** [comment]
@Fedik first post ist directly comparison vs J3 on same server. The different is extremly. Where Joomla 3 is not noticeable at all, joomla 4 strikes out extremely.

**Comment by HLeithner** [comment]
Do you have more matrices of the server load? I/O, memory usage, separated Info for Apache and SQL server. What's the timescale? Some more information about the hardware like baremetal or VM (which hypervisor). SSD, how they are connected nvme,SAS,sata?

**Comment by HRIT-Florian** [comment]
@Fedik same problem on mentioned PR 
@HLeithner dedicated Intel Xeon as mentioned in first post with 32 GB RAM on SSD SATA (not nvme)
memory usage does not strike out noticeably. More infos next...

**Comment by Fedik** [comment]
If it possible to run profiler (eg xdebug) and attach the result, that also would be nice.

**Comment by Fedik** [comment]
Here is Top 10 "slowest methods" from profiler from my test installation (php8, fpm, nfs). When open Dashboard.


Joomla 4:
![j4-default-top10](https://user-images.githubusercontent.com/1568198/114903300-56100f80-9e1f-11eb-91ca-ccc20cfd3bc6.png)


Joomla 3:
![j3-default-top10](https://user-images.githubusercontent.com/1568198/114903389-6fb15700-9e1f-11eb-9a95-dc6bc505b9af.png)

Joomla 3 is ~30% faster, If XDebug not lying.
When disable logging for Joomla 4 it become ~20% faster. (I mean make Log::add() empty :) )

**Comment by PhilETaylor** [comment]
I have several Joomla 4 sites on basic cheap $5 servers at DigitalOcean and Linode and I dont see any such CPU spikes, I also dont see any spikes locally.

**Comment by Fedik** [comment]
I suspect something with their file system set-up

**Comment by PhilETaylor** [comment]
Nic runs Joomla 4 for his live akeeba.com - Im sure if there were such (consistent, replicatable  ) spikes we would have heard about it by now ;-)

**Comment by brianteeman** [comment]
His blog not his main site ;)

I've been running it for a month on learnjoomla4.com and nothing but joy so far

**Comment by dgrammatiko** [comment]
sloth-theme.site (the only true SPA/PWA ever build with Joomla) runs J4 for a year on a $5 DO VPS and here are some stats
![Screenshot 2021-04-15 at 18 51 22](https://user-images.githubusercontent.com/3889375/114907851-cb79e100-9e1b-11eb-8fe2-2e9756677e51.png)

![Screenshot 2021-04-15 at 18 50 45](https://user-images.githubusercontent.com/3889375/114907883-d3d21c00-9e1b-11eb-8af0-b7bfba069778.png)

**Comment by alikon** [comment]
it's really hard to compare different metrics/tools
plus
each one of you have provided different "numbers/graph"

i'm inclined to close this issue...maybe room for https://github.com/joomla/joomla-cms/discussions

**Comment by HRIT-Florian** [comment]
Thanks all for your feedback. Our hoster has now started to check further with that issue. I'm with you, it seem not to be a Joomla issue. Anyway, if anything comes up, I will share the problem with you.

**Comment by MacJoom** [comment]
Closing discussion as no longer relevant and outdated.
